tealiumlibrary/src/main/java/com/tealium/core/network/HttpClient.kt

```diff
@@ -33,45 +33,50 @@ class HttpClient(var config: TealiumConfig,
             false
         } else true
 
-    override suspend fun post(payload: String, urlString: String, gzip: Boolean) = coroutineScope {
-        if (isActive && isConnected) {
-            try {
-                with(URL(urlString).openConnection() as HttpURLConnection) {
-                    try {
-                        doOutput = true
-                        setRequestProperty("Content-Type", "application/json")
-                        val dataOutputStream = when (gzip) {
-                            true -> {
-                                setRequestProperty("Content-Encoding", "gzip")
-                                DataOutputStream(GZIPOutputStream(outputStream))
-                            }
-                            false -> {
-                                DataOutputStream(outputStream)
+    override suspend fun post(payload: String, urlString: String, gzip: Boolean): Unit = coroutineScope {
+        withContext(Dispatchers.IO) {
+                try {
+                    with(URL(urlString).openConnection() as HttpURLConnection) {
+                        if (isActive && isConnected) {
+                            try {
+                                doOutput = true
+                                setRequestProperty("Content-Type", "application/json")
+                                val dataOutputStream = when (gzip) {
+                                    true -> {
+                                        setRequestProperty("Content-Encoding", "gzip")
+                                        DataOutputStream(GZIPOutputStream(outputStream))
+                                    }
+                                    false -> {
+                                        DataOutputStream(outputStream)
+                                    }
+                                }
+                                dataOutputStream.write(payload.toByteArray(Charsets.UTF_8))
+                                dataOutputStream.flush()
+                                dataOutputStream.close()
+                            } catch (e: Exception) {
+                                networkClientListener?.onNetworkError(e.toString())
                             }
+                            val code = responseCode
+                            val message = responseMessage
```

Comment:
Not major, but can `responseCode` and `responseMessage` be called inline with `onNetworkResponse()` call instead?

Reply:
sadly not - as the without calling either `responseCode` or `responseMessage` then the request will not complete.
if a `networkClientListener` has not been provided - as is the case on `ConsentManager`, those methods never get called and thus the data does not get sent